src/routes/process-user-vocabulary/components/ImageSelectionOverlay.tsx

```diff
@@ -33,39 +42,92 @@ const Container = styled.View`
 }
 
 const ImageSelectionOverlay = ({ setVisible, pushImage }: ImageSelectionOverlayProps): ReactElement => {
   const camera = useRef<RNCamera>(null)
+  const [showGallery, setShowGallery] = useState<boolean>(false)
+  const [photos, setPhotos] = useState<PhotoIdentifier[]>([])
+  const { permissionRequested, permissionGranted } = useGrantPermissions(
+    Platform.OS === 'ios' ? PERMISSIONS.IOS.MEDIA_LIBRARY : PERMISSIONS.ANDROID.READ_MEDIA_IMAGES
+  )
+
+  const getGalleryPictures = () => {
+    CameraRoll.getPhotos({ first: 30, assetType: 'Photos' })
+      .then(res => {
+        setPhotos(res.edges)
+        setShowGallery(true)
+      })
+      .catch(reportError)
+  }
+
+  const selectImage = async (imageUri: string) => {
+    // iOS saves images with an internal uri, react-native-fs has problems with uploading those files
+    if (imageUri.startsWith('ph://')) {
+      const compatibleUri = await CameraRoll.iosGetImageDataById(imageUri)
+      if (!compatibleUri.node.image.filepath) {
+        reportError('No filepath received from CameraRoll.iosGetImageDataById')
+        return
+      }
+      pushImage({ uri: compatibleUri.node.image.filepath, shouldBeCopied: true })
+    } else {
+      pushImage({ uri: imageUri, shouldBeCopied: true })
+    }
+    setVisible(false)
+  }
 
-  const takeImage = async () => {
+  const takePhoto = async () => {
     if (camera.current) {
       const options = { quality: 0.5, base64: true }
       const data = await camera.current.takePictureAsync(options)
       setVisible(false)
-      pushImage(data.uri)
+      pushImage({ uri: data.uri, shouldBeCopied: false })
     }
   }
 
   return (
     <CameraOverlay setVisible={setVisible}>
-      <Camera ref={camera} captureAudio={false} testID='camera'>
-        <Container>
-          <ActionBar>
-            {/* TODO LUN-440 implement gallery */}
-            {/* eslint-disable-next-line no-console */}
-            <PressableOpacity onPress={() => console.log('open gallery')}>
-              <ImageIcon width={GALLERY_ICON_SIZE} height={GALLERY_ICON_SIZE} testID='gallery-icon' />
-            </PressableOpacity>
-            <TakeImageButton onPress={takeImage}>
-              <CircleIconWhite width={TAKE_IMAGE_ICON_SIZE} height={TAKE_IMAGE_ICON_SIZE} testID='take-image-icon' />
-            </TakeImageButton>
-          </ActionBar>
-        </Container>
-      </Camera>
+      {showGallery ? (
+        <>
+          {permissionRequested && !permissionGranted && (
+            <NotAuthorisedView
+              description={getLabels().general.library.noAuthorization.description}
+              setVisible={setVisible}
+            />
+          )}
+          {permissionGranted && (
+            <ScrollView contentContainerStyle={{ flexDirection: 'row', flexWrap: 'wrap' }}>
+              {photos.map(photo => (
+                <Pressable key={photo.node.timestamp} onPress={() => selectImage(photo.node.image.uri)}>
```

Comment:
when i increased amount of images in the gallery, i got an duplicated key error. Not sure whats the reason for. Timestamps should be unique